api/turing/config/config.go

```diff
@@ -32,85 +40,100 @@ func (qty *Quantity) Decode(value string) (err error) {
 	return nil
 }
 
+// MarshalJSON implements the json.Marshaller interface so that we will get the expected
+// JSON string representation when marshalling a quantity object.
+func (qty *Quantity) MarshalJSON() ([]byte, error) {
+	q := resource.Quantity(*qty)
+	return q.MarshalJSON()
+}
+
 // Config is used to parse and store the environment configs
 type Config struct {
-	Port                int                  `envconfig:"turing_port" default:"8080"`
-	AllowedOrigins      []string             `split_words:"true" default:"*"`
-	AuthConfig          *AuthorizationConfig `envconfig:"authorization" validate:"required"`
-	DbConfig            *DatabaseConfig      `envconfig:"turing_database"`
-	DeployConfig        *DeploymentConfig    `envconfig:"deployment"`
-	RouterDefaults      *RouterDefaults      `envconfig:"turing_router"`
-	Sentry              sentry.Config        `split_words:"false" envconfig:"SENTRY"`
-	VaultConfig         *VaultConfig         `envconfig:"vault"`
-	TuringEncryptionKey string               `split_words:"true" required:"true"`
-	AlertConfig         *AlertConfig         `envconfig:"alert"`
-	MLPConfig           *MLPConfig
-	TuringUIConfig      *TuringUIConfig `envconfig:"turing_ui"`
+	Port                int `validate:"required"`
+	AllowedOrigins      []string
+	AuthConfig          *AuthorizationConfig
+	DbConfig            *DatabaseConfig   `validate:"required"`
+	DeployConfig        *DeploymentConfig `validate:"required"`
+	RouterDefaults      *RouterDefaults   `validate:"required"`
```

Comment:
General question: is it sufficient to have just a presence check (`required`) for config fields or do we need more thoughtful validation?

Reply:
No it's not sufficient. I'll leave this updated in another pull request.
https://github.com/gojek/turing/issues/25